### PR TITLE
Fix GNNExplainer with GCN

### PR DESCRIPTION
**Pull request by RBendias**
Currently, `GNNExplainer` uses the wrong number of hops in the GNNExplainer examples. The default value is equal to the number of message passing layers, but for an n-layer `GCN` model, the number of hops is equal to n+1.
Increasing `num_hops` by 1 would include too many edges in the explanation.
Instead, I excluded the normalization from each `GCNConv` and added `GCNNorm` as a data transformation.

### CODE REVIEW

File: examples/gnn_explainer.py [hunk 1]
```diff
@@ -9,15 +9,16 @@
...
 
 class Net(torch.nn.Module):
     def __init__(self):
         super(Net, self).__init__()
-        self.conv1 = GCNConv(dataset.num_features, 16)
-        self.conv2 = GCNConv(16, dataset.num_classes)
+        self.conv1 = GCNConv(dataset.num_features, 16, normalize=False)
+        self.conv2 = GCNConv(16, dataset.num_classes, normalize=False)
 
     def forward(self, x, edge_index):
```
Comment (rusty1s) [inline examples/gnn_explainer.py:23]:
```suggestion
    def forward(self, x, edge_index, edge_weight):
```

File: examples/gnn_explainer_ba_shapes.py [hunk 2]
```diff
@@ -18,7 +19,7 @@
 device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
 data = data.to(device)
 model = GCN(data.num_node_features, hidden_channels=20, num_layers=3,
-            out_channels=dataset.num_classes).to(device)
+            out_channels=dataset.num_classes, normalize=False).to(device)
```
[the remark this answers is not in the document]
Comment (RBendias) [reply examples/gnn_explainer_ba_shapes.py:22]: I see, I'll adjust it.

**Review by rusty1s — APPROVED** [review]
LGTM. Thank you.
